src/common/event-emitter.ts

```diff
@@ -52,12 +48,14 @@ export class EventEmitter<D extends [...any[]]> {
   }
 
   emit(...data: D) {
-    [...this.listeners].every(([callback, options]) => {
-      if (options.once) {
+    for (const [callback, { once }] of this.listeners) {
+      if (once) {
         this.removeListener(callback);
       }
 
-      return callback(...data) !== false;
-    });
+      if (callback(...data) === false) {
```

Comment:
This reads like stop calling rest of the callback if one of the callback returns false, doesn't seems be the same before refactoring?

Reply:
It is what the previous had, the docs of every say so.